server/opendp_apps/dataverses/testing/test_endpoints.py

```diff
@@ -0,0 +1,122 @@
+import json
+from unittest import skip
+import tempfile
+import requests_mock
+
+from django.test import Client, tag, TestCase
+from django.urls import reverse
+
+from django.conf import settings
+from django.template.loader import render_to_string
+
+from django.contrib.auth import get_user_model
+from opendp_apps.dataverses import static_vals as dv_static
+from opendp_apps.dataverses.models import ManifestTestParams, DataverseHandoff, RegisteredDataverse
+from opendp_apps.dataverses.dataverse_manifest_params import DataverseManifestParams
+from opendp_apps.dataverses.dataverse_request_handler import DataverseRequestHandler
+from opendp_apps.user.models import DataverseUser, OpenDPUser
+from opendp_apps.dataset.models import DataverseFileInfo
+from opendp_apps.model_helpers.msg_util import msgt
+
+TAG_WEB_CLIENT = 'web-client' # skip these tests on travis; need to fix as many use requests to access the localhost
+
+
+class BaseEndpointTest(TestCase):
+
+    fixtures = ['test_dataverses_01.json',
+                'test_manifest_params_04.json',
+                # This file needs a real token in order to run tests
+                'test_dataverse_handoff_01.json',
+                'test_opendp_users_01.json',
+                'test_dataverse_user_01.json']
+
+    def setUp(self):
+
+        # test client
+        self.client = Client()
+
+        self.user_obj, _created = get_user_model().objects.get_or_create(username='dv_depositor')
+
+        self.client.force_login(self.user_obj)
+
+        self.mock_params = ManifestTestParams.objects.filter(use_mock_dv_api=True).first()
+
+
+class DataversePostTest(BaseEndpointTest):
+
+    def test_successful_creation(self):
```

Comment:
Right! I see what you're saying about testing with a live Dataverse and tokens. I'll plan to do some work to revise the "mock dataverse" urls.